src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisonperson/profiledetails/ProfileDetailsService.kt

```diff
@@ -114,12 +115,15 @@ class ProfileDetailsService(
   private fun getProfileType(profileType: String) = profileTypeRepository.findByIdOrNull(profileType)
     ?: throw BadDataException("Invalid profile type $profileType")
 
-  private fun validateProfileCode(profileType: String, profileCode: String?) =
-    profileCode?.let {
-      if (!profileCodeRepository.existsById(ProfileCodeId(profileType, profileCode))) {
-        throw BadDataException("Invalid profile code $profileCode for profile type $profileType")
+  private fun validateProfileCode(profileType: String, profileCode: String?) {
+    profileCode
+      ?.takeIf { getProfileType(profileType) is ProfileTypeCode }
+      ?.also {
+        if (!profileCodeRepository.existsById(ProfileCodeId(profileType, profileCode))) {
+          throw BadDataException("Invalid profile code $profileCode for profile type $profileType")
+        }
```

Comment:
The change is to only validate a profile code if the NOMIS column has `CODE_VALUE_TYPE=CODE`. That column is used as a discriminator for type `ProfielTypeCode`.

Also removed the return type, this is validation by exception so not needed.